solutions/swb-app/src/costCenterRoutes.ts

```diff
@@ -23,6 +27,30 @@ export function setUpCostCenterRoutes(router: Router, costCenterService: CostCen
     })
   );
 
+  router.put(
+    '/costCenters/:id/softDelete',
+    wrapAsync(async (req: Request, res: Response) => {
+      const deleteCostCenterRequest = { id: req.params.id };
+      const validatedRequest = validateAndParse<DeleteCostCenterRequest>(
+        DeleteCostCenterRequestParser,
+        deleteCostCenterRequest
+      );
+      res.status(204).send(await costCenterService.softDeleteCostCenter(validatedRequest));
```

Comment:
As discussed offline, this should be a 204 because the user just needs to know whether the resource was deleted or not. We don't need to return the underlying details of how we implement soft delete. In our case our implementation of soft delete is not the standard approach and will confuse user if we return the object.